src/LightSaml/Model/XmlDSig/SignatureXmlReader.php

```diff
@@ -60,17 +61,15 @@ public function getSignature()
     /**
      * @return bool
      *
-     * @throws LightSamlSecurityException
+     * @throws LightSamlSecurityException|Exception
      */
     public function validate(XMLSecurityKey $key)
     {
         if (null == $this->signature) {
             return false;
         }
 
-        if (false == $this->signature->validateReference()) {
```

Comment:
Hello, thx for your contribution.

Can you handle validation exception in try catch and throw a LightSamlSecurityException, like this :

```php
        try {
            $this->signature->validateReference();
        } catch (Exception $e) {
            throw new LightSamlSecurityException('Digest validation failed', $e->getCode(), $e);
        }
```

Reply:
Changed.
I didn't remove Exception from @throws because it can still be thrown and it's covered in the example, https://github.com/lightSAML/lightSAML/blob/master/examples/how_to_verify_signature_of_saml_message.php   : 
`} catch (\Exception $ex) {`

Next method call, [$this->castKeyIfNecessary](https://github.com/lightSAML/lightSAML/blob/master/src/LightSaml/Model/XmlDSig/SignatureXmlReader.php#L75)  , uses [KeyHelper::castKey](https://github.com/lightSAML/lightSAML/blob/master/src/LightSaml/Credential/KeyHelper.php#L58) , which does `$newKey->loadKey` than can also throw Exception.  ( see https://github.com/robrichards/xmlseclibs/blob/master/src/XMLSecurityKey.php#L357 )

Next is method $this->signature->verify  which also can throw Exception.
( see https://github.com/robrichards/xmlseclibs/blob/master/src/XMLSecurityDSig.php#L812 ) 

If these should be changed to bubble up LightSamlException from Exception let me know, will do separate PRs for them.

Reply:
Ok, very nice.